### PR TITLE
Enforce "resource" field in request objects.

### PR DESCRIPTION
**Pull request by kuba**
Corresponds to:
- https://github.com/letsencrypt/boulder/pull/442
- https://github.com/letsencrypt/acme-spec/pull/156

r? @jsha? @jdkasten?

### CODE REVIEW

**Comment by jdkasten** [comment]
:+1:  LGTM

File: acme/messages.py [hunk 1]
```diff
@@ -137,13 +137,15 @@ class ResourceBody(jose.JSONObjectWithFields):
...
 
-class Registration(ResourceBody):
+class Registration(interfaces.ClientRequestableResource, ResourceBody):
     """Registration Resource Body.
 
     :ivar acme.jose.jwk.JWK key: Public key.
     :ivar tuple contact: Contact information following ACME spec
 
     """
+    resource_type = 'new-regr'
```
Comment (jsha) [inline acme/messages.py]: Typo: this should be new-reg, not new-regr.

Comment (jdkasten) [inline acme/messages.py]: Thanks! Fixing now...

Comment (kuba) [inline acme/messages.py]: Well spotted, thanks!

Comment (kuba) [inline acme/messages.py]: Sorry, @jdkasten, just noticed... Fixed in #596.

**Pull request merged**